pkg/kine/server/get.go

```diff
@@ -23,11 +23,9 @@ func (l *LimitedServer) get(ctx context.Context, r *etcdserverpb.RangeRequest) (
 		attribute.Int64("revision", r.Revision),
 	)
 
-	if len(r.RangeEnd) != 0 {
-		return nil, fmt.Errorf("unexpected rangeEnd: want empty, got %s", r.RangeEnd)
-	}
-	if r.Limit != 0 {
-		return nil, fmt.Errorf("unexpected limit: want 0, got %d", r.Limit)
+	if r.Limit != 0 && len(r.RangeEnd) != 0 {
+		err := fmt.Errorf("invalid combination of rangeEnd and limit, limit should be 0 got %d", r.Limit)
```

Comment:
just curious here: This basically enforces no limit on a range request, right?
Why can't we have a range request with a limit which would just return the first X entries?